### DISCUSSION TITLE
Port LabelSizeFontToFit effext from Xamarin Toolkit

### DISCUSSION THREAD

**Discussion by TFreudi1** [CommunityToolkit/Maui#876] [General]
I have Xamarin projects that I want to port to MAUI that make use of LabelSizeFontToFit effect from the xamarin toolkit.
I don't find a similiar behaviour in current MAUI Toolkit.
The MAUI team itself move a similiar feature from milestone to milestone :-(

**Comment by pictos** [comment]
@TFreudi1 can you link this feature here, I couldn't find it on Xamarin Community Toolkit

**Reply by TFreudi1** [reply]
You are right, it was removed in the current version, I use an older Nuget Package, I use it like that:
<Label.Effects>
  <effects:LabelSizeFontToFit />
</Label.Effects>

**Reply by pictos** [reply]
Do you have a link to the source code?

**Reply by TFreudi1** [reply]
Not really because I use an old Nuget Packge but check this: 
https://github.com/xamarin/XamarinCommunityToolkit/issues/60

**Reply by pictos** [reply]
Thanks @TFreudi1, for reference [here's the original implementation](https://github.com/xamarin/XamarinCommunityToolkit/tree/ddae5d9b1b4dd7ab903c9c3ebe869781835c320d/Toolkit/Effects/Label)

If you want this on toolkit, feel free to open a new discussion with the API Spec for this feature, that way we can vote and implement it.

**Reply by TFreudi1** [reply]
Well ok, It will be the same like this:
[Feature Request MAUI](https://github.com/dotnet/maui/issues/10884)

**Comment by IeuanWalker** [comment]
A feature like this was almost added to Xamarin, but was closed because it was a new feature and MAUI was the focus. But I've created this issue in the MAUI repo - https://github.com/dotnet/maui/issues/10884

**Comment by lmoe** [comment]
@TFreudi1 I have the same trouble. I'm currently working on a widget based dashboard and want them to be resizable. The label fontsize is not responsive, so it looks very odd. I don't want to polute the ViewModel with OnLayout change events just to update the font size.

I saw that the MauiToolkit is having an OnScreenSize behavior at some point, which can at least handle different screen sizes but that doesn't help really much in this case.

I wrote a small behaviour that can attach to another controls property (width/height..) and size the font width with some multiplier. I rarely do any frontend work and have almost no skills with Xaml and such, so I don't know how good that actually is. But for me it gets the job done until the SizeFontToFit feature is there. Considering, it will be there in MAUI at some point.

You can use it if you like. 
Screenshot: https://imgur.com/a/ADkwY5c
https://pastebin.com/DMmLDv5d

Use it like so:
```           
<Label 
    Grid.Column="1" 
    Grid.Row="0" 
    TextColor="{DynamicResource PrimaryColor}" 
    Text="{Binding FriendlyName}" 
>   
    <Label.Behaviors>
        <behaviors:OnControlSize 
            ConstraintValue="{Binding Source={x:Reference ContainerGrid}, Path=Width}" 
            MinFontSize="1" 
            MaxFontSize="50" 
            FontSizeMultiplicator="0.06"  
        />
    </Label.Behaviors>
</Label>
```

Might be useful to put the actual font spacing/width into consideration or some other things I currently have no idea about. :D

**Reply by TFreudi1** [reply]
Thanks Imoe, but in your solution I missed the part that checks if the text will fit the label. I m also not a specialist for GUI but after googling a lot my guess is that it must be someting wie OnMeasure. My problem is that I have 4 GridColumns defined with 20*,24* ... and fixed Text in some of the columns, for example "Name:" Now on some mobiles (different displays/resolutions)  it fits the griddcell, but on other you will have a line break, like:
Nam
e:
But because that linebreaks happen, the system must know that the text with the choosed font won't fit the cell. And that part I'm interested in, or better the Label will have a bool property "LabelSizeFontToFit".

**Comment by CliffAgius** [comment]
Closed as per the Community Stand-up discussion here: https://youtu.be/N9wMcBP4jtg?t=2889
